src/pint/models/timing_model.py

```diff
@@ -2534,6 +2619,176 @@ def __init__(self):
         self.phase_derivs_wrt_delay = []
 
 
+class AllComponents:
+    """ A class for the components pool.
+
+    This object stores and manages the instances of component classes with class
+    attribute .register = True. This includes the PINT built-in components and
+    user defined components. It is designed for helping model building and
+    parameter seraching, not for directly data analysis. Thus, the instances do
+    not have any valid parameter values. Runing `.validate()` function will fail.
```

Comment:
Please explain why this class stores these half-formed invalid instances. I believe it is because classes do not know about their parameters until an instance is created, and this object needs to obtain that information?